xds/internal/balancer/ringhash/ringhash.go

```diff
@@ -180,7 +180,18 @@ type ringhashBalancer struct {
 
 	config *LBConfig
 
-	subConns map[resolver.Address]*subConn // `attributes` is stripped from the keys of this map (the addresses)
+	// The key for this map is a resolver.Address with the following
+	// modification:
+	//  - `Attributes` field is cleared and rewritten with a single attribute
+	//     containing the weight of the address. The `AddressMap` type uses the
+	//     `Attributes` field to determine equality, but ignores the
+	//     `BalancerAttributes` field. Hence, we copy over the weight of the
+	//     address from the latter to the former.
```

Comment:
```suggestion
	//  - `Attributes` field is cleared and rewritten with a single attribute
	//    containing the weight of the address. The `AddressMap` type uses the
	//    `Attributes` field to determine equality, but ignores the
	//    `BalancerAttributes` field. Hence, we copy over the weight of the
	//    address from the latter to the former.
```

Reply:
Minor nit; not sure if this "suggestion" thing works & makes it easy to apply, but that's the goal.